File: ios/Sasquatsh/Views/Shared/D20ProgressView.swift
import SwiftUI

/// Drop-in replacement for ProgressView() using the spinning d20
struct D20ProgressView: View {
    var size: CGFloat = 40
    var message: String? = nil

    var body: some View {
        VStack(spacing: 8) {
            D20SpinnerView(size: size)
                .frame(width: size, height: size)
            if let message {
                Text(message)
                    .font(.md3BodyMedium)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
            }
        }
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/shared/D20ProgressView.kt
package com.sasquatsh.app.views.shared

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.size
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Drop-in replacement for CircularProgressIndicator using a spinning D20 die.
 * Port of the iOS D20ProgressView.
 */
@Composable
fun D20ProgressView(
    modifier: Modifier = Modifier,
    size: Dp = 40.dp,
    message: String? = null
) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        D20SpinnerView(
            modifier = Modifier.size(size),
            size = size
        )

        if (message != null) {
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = message,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
    }
}
